extensions/kafka-client/runtime/src/test/java/io/quarkus/kafka/client/runtime/KafkaRuntimeConfigProducerTest.java

```diff
@@ -0,0 +1,86 @@
+        when(appConfig.name()).thenReturn(Optional.empty());
+
+        KafkaRuntimeConfigProducer producer = new KafkaRuntimeConfigProducer();
+        Map<String, Object> result = producer.createKafkaRuntimeConfig(config, appConfig);
+
+        // Verify that the tls-configuration-name is present
+        assertEquals("my-tls-config", result.get("tls-configuration-name"));
+        // Verify that ssl.engine.factory.class is set
+        assertEquals(QuarkusKafkaSslEngineFactory.class.getName(), result.get("ssl.engine.factory.class"));
+    }
+
+    @Test
+    public void testTlsConfigurationNameWithUnderscores() {
+        // Test when property is set via environment variable as KAFKA_TLS_CONFIGURATION_NAME
+        Config config = new SmallRyeConfigBuilder()
+                .withDefaultValues(Map.of("KAFKA_TLS_CONFIGURATION_NAME", "my-tls-config"))
+                .build();
+        ApplicationConfig appConfig = mock(ApplicationConfig.class);
+        when(appConfig.name()).thenReturn(Optional.empty());
+
+        KafkaRuntimeConfigProducer producer = new KafkaRuntimeConfigProducer();
+        Map<String, Object> result = producer.createKafkaRuntimeConfig(config, appConfig);
+
+        // After processing, underscores are converted to dots
+        assertEquals("my-tls-config", result.get("tls-configuration-name"));
+        // Verify that ssl.engine.factory.class is set (this is the fix for #51195)
+        assertEquals(QuarkusKafkaSslEngineFactory.class.getName(), result.get("ssl.engine.factory.class"));
+    }
+
+    @Test
+    public void testOtherKafkaProperties() {
+        // Test that other Kafka properties are processed correctly
+        Config config = new SmallRyeConfigBuilder()
+                .withDefaultValues(Map.of("kafka.bootstrap.servers", "localhost:9092",
+                        "KAFKA_CONSUMER_MAX_POLL_RECORDS", "100"))
+                .build();
+        ApplicationConfig appConfig = mock(ApplicationConfig.class);
+        when(appConfig.name()).thenReturn(Optional.empty());
+
+        KafkaRuntimeConfigProducer producer = new KafkaRuntimeConfigProducer();
+        Map<String, Object> result = producer.createKafkaRuntimeConfig(config, appConfig);
+
+        assertEquals("localhost:9092", result.get("bootstrap.servers"));
+        assertEquals("100", result.get("consumer.max.poll.records"));
+    }
+
+    @Test
+    public void testGroupIdDefaultsToApplicationName() {
+        SmallRyeConfig config = new SmallRyeConfigBuilder()
+                .withDefaultValues(Map.of("kafka.bootstrap.servers", "localhost:9092"))
+                .build();
+        ApplicationConfig appConfig = mock(ApplicationConfig.class);
+        when(appConfig.name()).thenReturn(Optional.of("my-app"));
+
+        KafkaRuntimeConfigProducer producer = new KafkaRuntimeConfigProducer();
+        Map<String, Object> result = producer.createKafkaRuntimeConfig(config, appConfig);
+
+        assertEquals("my-app", result.get("group.id"));
```

Comment:
Here is a suggestion to give a better assertion message: 

```suggestion
        assertThat(result).containsEntry("group.id", "my-app");
```